STYLE_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+
+We recommend checking [axe](https://www.deque.com/axe/) and [Lighthouse](https://developers.google.com/web/tools/lighthouse) to review for potential accessibility errors if unsure.
+
+## Semantic HTML
+
+Ensure that page content is marked up with [semantic HTML elements](https://html.com/semantic-markup/) in order to ensure that it is as usable and accessible as possible.
+
+Avoid:
+```jsx
+<div className="form">
+  <div>Contact us</div>
+  <div>Use this form to send us a message.</div>
+  <div>Email address:</div>
+  <input type="text" />
+  <div onClick={handleSubmit}>Submit</div>
+</div>
+```
+Prefer:
+```jsx
+<form onSubmit={handleSubmit}>
+  <h1>Contact us</h1>
+  <p>Use this form to send us a message.</p>
+  <label htmlFor="email">Email address:</label>
+  <input id="email" type="email" />
+  <button>Submit</button>
+</form>
+```
+
+## CSS
+
+### Class names
+
+This project uses the [BEM naming convention](http://getbem.com/), to avoid conflicts, reduce rule specificity, and make classes easier to read and understand. [Reducing rule specificity](https://css-tricks.com/strategies-keeping-css-specificity-low/) is particularly important for improving CSS maintainability, so we try to avoid nesting CSS selectors as much as possible.
+
+We use the `pipeline-` prefix for most HTML/CSS classes, to reduce the risk of clashes in the global namespace when this project is imported into other applications. It's important never 
+
+<!-- @TODO Explain usage of the 'kedro' class, Kedro-UI, and webfonts -->
+
+### Variables
+
+We use Sass variables to declare colours and reused values wherever useful. It's not necessary to declare every CSS unit value as a variable, but if a value is used in multiple places (for related reasons), then it makes sense to make it a variable - either locally in the component, or application-wide in `src/styles/_variables.scss`. Using a named variable instead of a number can clarify your intent, and help you avoid having to update a value in multiple places later on.
+
+Avoid:
+```scss
+color: #001521;
+color: rgba(#001521, 0.5);
+color: rgba(0, 0, 0, 0.5);
+```
+
+Prefer:
+```scss
+color: $color-bg-dark-1;
+color: rgba($color-bg-dark-1, 0.5);
+color: rgba(black, 0.5);
+```
+
+### Theme colours
+
+Kedro-Viz has both light and dark themes, and every component must be styled appropriately for each theme. The themes are set using the same theme classes that Kedro-UI uses - `kui-theme--light` and `kui-theme--dark` - applied on the top-level container element. In order to simplify code and avoid increasing CSS specificity, we use CSS Custom Properties to handle theme colours, and we recommend that you use these where possible.
```

Comment:
```suggestion
Kedro-Viz has both light and dark themes, and every component must be styled appropriately for each theme. The themes are set using the same theme classes that Kedro-UI uses - `kui-theme--light` and `kui-theme--dark` - applied on the top-level container element. To simplify code and avoid increasing CSS specificity, we use CSS Custom Properties to handle theme colours. We recommend that you use these too where possible.
```

Reply:
Can you mandate it? Should it be more imperative? Also, do you think it's necessary to point to a link about custom properties? Do they also need to be capitalised as a proper noun? So many questions...

Reply:
You're right: "Please use these wherever possible" is better.

[Custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) are a core but new-ish part of CSS, so they shouldn't need linking to, but we can. I'd mainly seen them capitalised in headings but you're right, they don't need capitalisation. I'll change that.